module/src/clients/surface_temp.c

```diff
@@ -173,6 +194,24 @@ static int ssam_temp_probe(struct ssam_device *sdev)
 	if (!ssam_temp)
 		return -ENOMEM;
 
+	// Retrieve the name for each sensor.
+	for (channel = 0; channel < SSAM_TMP_SENSOR_MAX_COUNT; channel++)
+	{
+		if ((sensors & BIT(channel)) == 0)
+			continue;
+
+		status =  ssam_retry(__ssam_tmp_get_name, sdev->ctrl,
```

Comment:
Is the `ssam_retry` needed here? There were some functions in the past that would sporadically fail repeatedly on some devices (like extended battery info on the SB2, likely because they returned a lot of data), which is why this macro was added. But generally, this does re-trying on top of the internal re-trying that already happens. And I think on newer devices, communication also seems to be much more stable. So I'd prefer to avoid this.

Reply:
Probably not, it was not added because it didn't work otherwise at least: I copied the retry from the [surface_platform_profile.c](https://github.com/linux-surface/kernel/blob/8cd23b1bb3a8b7a3ef7cec2c37e7e46e6397a858/drivers/platform/surface/surface_platform_profile.c#L50).